test/correctness/simd_op_check.cpp

```diff
@@ -1838,6 +1846,11 @@ struct Test {
         check("v*.h += vmpy(v*.ub,v*.b)", hvx_width/1, i16_1 + i16(i8_1) * i16(u8_2));
         check("v*.w += vmpy(v*.h,v*.uh)", hvx_width/2, i32_1 + i32(u16_1) * i32(i16_2));
 
+        if(is_v65) {
+           check("v*.w += vmpy(v*.h, r*.h)", hvx_width/1, i32_1 + i32(i16_1)*32767);
```

Comment:
I think these are probably falsely passing right now, because the generated code probably has something that matches this pattern, but it also has ":sat" following it...

Reply:
Just kept the test case as it is. Removed the pattern matching from HexagonOptimize as it was redundant. Or should this test case be removed too?

Reply:
I think this should not be inside an if(is_v65) branch now that there's nothing v65 specific about the code that handles these patterns. I think the actual code will have a ":sat" suffix on these instructions though.